central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -84,88 +92,151 @@ type ResultEmail struct {
+	clusters := req.ClusterIDs
+	resultsCSV := make(map[string][]*ResultRow)
+	resultEmailComplianceReport := &ResultEmail{
+		TotalPass:  0,
+		TotalMixed: 0,
+		TotalFail:  0,
+		Clusters:   0,
+	}
 
-	return nil, nil
+	for _, clusterID := range clusters {
+		parsedQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanConfig, req.ScanConfigID).
+			AddExactMatches(search.ClusterID, clusterID).
+			ProtoQuery()
+		resultCluster := []*ResultRow{}
+
+		err := rg.checkResultsDS.WalkByQuery(req.Ctx, parsedQuery, func(c *storage.ComplianceOperatorCheckResultV2) error {
+			row := &ResultRow{
+				ClusterName: c.GetClusterName(),
+				CheckName:   c.GetCheckName(),
+				Description: c.GetDescription(),
+				Status:      c.GetStatus().String(),
+				Remediation: c.GetInstructions(),
+			}
+			// get profile name for the check result
+			q := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanRef, c.GetScanRefId()).ProtoQuery()
+			log.Info("query to get profiles is %s", q)
+			profiles, err := rg.profileDS.SearchProfiles(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(profiles) < 1 {
+				return errors.New("Profile not found")
+			}
+
+			row.Profile = profiles[0].GetName()
+
+			// get remediation for the check result
+			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorCheckName, c.GetCheckName()).AddExactMatches(search.ClusterID, c.GetClusterId()).ProtoQuery()
+			remediations, err := rg.remediationDS.SearchRemediations(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(remediations) > 0 {
+				row.Remediation = remediations[0].GetName()
+			}
+
+			resultCluster = append(resultCluster, row)
+			if c.GetStatus() == storage.ComplianceOperatorCheckResultV2_FAIL {
+				resultEmailComplianceReport.TotalFail += 1
+			} else if c.GetStatus() == storage.ComplianceOperatorCheckResultV2_PASS {
+				resultEmailComplianceReport.TotalPass += 1
+			} else {
+				resultEmailComplianceReport.TotalMixed += 1
+			}
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		resultsCSV[clusterID] = resultCluster
+	}
+	resultEmailComplianceReport.Clusters = len(req.ClusterIDs)
+	resultEmailComplianceReport.Profiles = req.Profiles
+	resultEmailComplianceReport.ResultCSVs = resultsCSV
+	return resultEmailComplianceReport, nil
 }
 
 func (rg *complianceReportGeneratorImpl) sendEmail(ctx context.Context, zipData *bytes.Buffer, emailBody *formatBody, formatEmailSub *formatSubject, notifiersList []*storage.NotifierConfiguration) {
 
 	errorList := errorhelpers.NewErrorList("Error sending compliance report email notifications")
-	for _, notifier := range notifiersList {
-		nf := rg.notificationProcessor.GetNotifier(ctx, notifier.GetEmailConfig().GetNotifierId())
+	for _, repNotifier := range notifiersList {
+		notifierID := repNotifier.GetId()
+		nf := rg.notificationProcessor.GetNotifier(ctx, notifierID)
```

Comment:
nit since I don't think you are using `notifierID` anywhere else.
```suggestion
		nf := rg.notificationProcessor.GetNotifier(ctx, repNotifier.GetId())
```